views/biobank/order.html.twig

```diff
@@ -88,23 +88,42 @@
                 {% include 'partials/biobank-order-details.html.twig' with { type: 'collected' } %}
             </div>
             <div role="tabpanel" class="tab-pane" id="process">
-                {% if order.collected_ts is not empty %}
-                    {% include 'partials/biobank-order-details.html.twig' with { type: 'processed' } %}
-                {% endif %}
+                {% include 'partials/biobank-order-details.html.twig' with { type: 'processed' } %}
             </div>
             <div role="tabpanel" class="tab-pane" id="finalize">
-                {% if order.collected_ts is not empty and order.processed_ts is not empty %}
-                    {% include 'partials/biobank-order-details.html.twig' with { type: 'finalized' } %}
-                {% endif %}
+                {% include 'partials/biobank-order-details.html.twig' with { type: 'finalized' } %}
             </div>
         </div>
     </div>
 {% endblock %}
 {% block pagejs %}
     <script>
         $(document).ready(function() {
+            var currentStep = {{ currentStep|json_encode|raw }};
+            if ($('.finalize-form .alert-danger').length !== 0) {
+                currentStep = 'finalize';
+            }
             // Switch tab to active step
-            $(".nav-tabs a[href='#{{ currentStep }}']").tab('show');
+            $(".nav-tabs a[href='#" + currentStep + "']").tab('show');
+
+            $('#checkall').on('change', function() {
+                $('#form_finalized_samples input:checkbox:enabled').prop('checked', $(this).prop('checked'));
+            });
+
+            $('#form_finalized_ts').pmiDateTimePicker();
+
+            $('.finalize-form button[type="submit"]').on('click', function () {
+                //Display warning message
+                var message = 'Are you sure you want to finalize this order?';
+                $('input[name="form[finalized_samples][]"]').each(function () {
+                    //Select samples that are unchecked and not disabled
+                    if ($(this).prop('checked') === false && $(this).prop('disabled') === false) {
+                        message = 'Warning: At least one sample that was collected and processed (as applicable) was not finalized. Are you sure you wish to continue?';
```

Comment:
We might want to check to see if we want this warning to apply to the biobank users.